ext/storage_sources/s3_store/CMakeLists.txt

```diff
@@ -1,6 +1,18 @@
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 project(s3_store CXX)
 
+# On certain platforms with older compiler versions, an additional library needs to be linked in order for the std::filesystem include to work.
+# See the notes section of the documentation below for details.
+# https://en.cppreference.com/w/cpp/filesystem
+# This ultimately will need to be modified to be compiler dependent instead of platform dependent.set(filesystem_lib)
+set(filesystem_lib)
+if (WT_LINUX)
+    set(filesystem_lib stdc++fs)
+endif()
```

Comment:
In addition, any additional library that needs to be linked I believe is compiler and compiler version dependent. This'll need to be adjusted too. See notes at the bottom of [this page](https://en.cppreference.com/w/cpp/filesystem).